### PR TITLE
add test to verify binaries are signed on release branches

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

This creates a cirrus test that will run on release branches and verify that the engine binaries are codesigned on macos.

## Related Issues

This test would prevent issues like https://github.com/flutter/flutter/issues/50066.

## Tests

I added an integration test file `dev/bots/codesign.dart` that verifies that certain cache binaries are codesigned.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

### CODE REVIEW

**Comment by Piinks** [comment]
#49917 is happening here prior to #50088

File: dev/bots/codesign.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  final String binaryPath;
+  final String stderr;
+}
+
+void main() {
+  // TODO(fujino): parse codesigning manifest for all files to verify
+  final List<String> binaries = <String>[
+    path.join('bin', 'cache', 'dart-sdk', 'bin', 'dart'),
+    path.join('bin', 'cache', 'artifacts', 'engine', 'darwin-x64', 'flutter_tester'),
+    path.join('bin', 'cache', 'artifacts', 'engine', 'darwin-x64', 'gen_snapshot'),
```
Comment (dnfield) [inline dev/bots/codesign.dart]: font-subset?

Do we need to verify the ios-deploy and imobiledevice are signed asd well?

Comment (christopherfujino) [reply dev/bots/codesign.dart]: I was gonna (in a future PR) have this script parse the actual manifest file of all binaries in the engine to sign, but now I realize that that won't be easily accessible from the framework. I will hard-code them all here.

Comment (dnfield) [reply dev/bots/codesign.dart]: Talked offline - it should be possible to just find all the binaries in bin/cache with the right MIME type and check those.

**Review by dnfield — APPROVED** [review]
LGTM!

**Comment by dnfield** [comment]
I think this would be worth cherry picking onto the hot fix branch - both to test it out and to make sure we don't end up releasing a hotfix without this.

File: .cirrus.yml [hunk 2]
```diff
@@ -652,6 +652,15 @@ task:
         - ulimit -S -n 2048 # https://github.com/flutter/flutter/issues/2976
         - ./dev/bots/deploy_gallery.sh
 
+    - name: verify_binaries_codesigned-macos # macos-only
+      # TODO(fujino): remove this `only_if` after https://github.com/flutter/flutter/issues/44372
+      only_if: "$CIRRUS_BRANCH == 'dev' || $CIRRUS_BRANCH == 'beta' || $CIRRUS_BRANCH == 'stable'"
```
Comment (dnfield) [inline .cirrus.yml]: Late add - we should make this run on hotfix branches too.

Comment (christopherfujino) [reply .cirrus.yml]: Oh yeah, that'd be nice. Can you do pattern matching in the `.cirrus.yml` file @fkorotkov?

**Comment by christopherfujino** [comment]
> I think this would be worth cherry picking onto the hot fix branch - both to test it out and to make sure we don't end up releasing a hotfix without this.

Yeah, that's my plan.

File: .cirrus.yml (hunk 2 above)
Comment (fkorotkov) [reply .cirrus.yml]: [Yes](https://cirrus-ci.org/guide/writing-tasks/#supported-operators), you can do something like `$CIRRUS_BRANCH =~ '.*hotfix.*'`

Comment (christopherfujino) [reply .cirrus.yml]: @fkorotkov thanks!

**Review by dnfield — COMMENTED** [review]
L even better TM

File: dev/bots/codesign.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      '+111', // is executable
+    ],
+  );
+  final List<String> allFiles = (result.stdout as String).split('\n').where((String s) => s.isNotEmpty).toList();
+  return allFiles.where(isBinary).toList();
+}
+
+void main() {
+  final List<String> failures = <String>[];
+
```
Comment (jmagman) [inline dev/bots/codesign.dart]: Should we run `flutter precache --no-android` to make sure the iOS engine artifacts are actually present?

**Pull request merged**